{"cells":[{"cell_type":"markdown","source":["# **SET UP**"],"metadata":{"id":"TNxbyNeKzb6o"},"id":"TNxbyNeKzb6o"},{"cell_type":"markdown","source":["## **Install**"],"metadata":{"id":"nFDBIKb50X1e"},"id":"nFDBIKb50X1e"},{"cell_type":"code","source":["from google.colab import drive\n","drive.mount('/content/drive')"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":304},"id":"jgav-LSGoxQp","outputId":"dcf6f08c-f14b-4075-f8a1-4235565349fa","executionInfo":{"status":"error","timestamp":1727709424362,"user_tz":-420,"elapsed":123223,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"jgav-LSGoxQp","execution_count":1,"outputs":[{"output_type":"error","ename":"ValueError","evalue":"mount failed","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)","\u001b[0;32m<ipython-input-1-d5df0069828e>\u001b[0m in \u001b[0;36m<cell line: 2>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mgoogle\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolab\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mdrive\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mdrive\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmount\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/content/drive'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m","\u001b[0;32m/usr/local/lib/python3.10/dist-packages/google/colab/drive.py\u001b[0m in \u001b[0;36mmount\u001b[0;34m(mountpoint, force_remount, timeout_ms, readonly)\u001b[0m\n\u001b[1;32m     98\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mmount\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmountpoint\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mforce_remount\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout_ms\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m120000\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreadonly\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     99\u001b[0m   \u001b[0;34m\"\"\"Mount your Google Drive at the specified mountpoint path.\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 100\u001b[0;31m   return _mount(\n\u001b[0m\u001b[1;32m    101\u001b[0m       \u001b[0mmountpoint\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    102\u001b[0m       \u001b[0mforce_remount\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mforce_remount\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.10/dist-packages/google/colab/drive.py\u001b[0m in \u001b[0;36m_mount\u001b[0;34m(mountpoint, force_remount, timeout_ms, ephemeral, readonly)\u001b[0m\n\u001b[1;32m    285\u001b[0m             \u001b[0;34m'https://research.google.com/colaboratory/faq.html#drive-timeout'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    286\u001b[0m         )\n\u001b[0;32m--> 287\u001b[0;31m       \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'mount failed'\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mextra_reason\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    288\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mcase\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m4\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    289\u001b[0m       \u001b[0;31m# Terminate the DriveFS binary before killing bash.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mValueError\u001b[0m: mount failed"]}]},{"cell_type":"code","source":["import os\n","os.chdir('/content/drive/My Drive/Colab Notebooks/BIOINFORMATICS_LECTURE/SD4108/PERTEMUAN/COLAB_version')"],"metadata":{"id":"1AjM2Gz0qTnE","executionInfo":{"status":"aborted","timestamp":1727709424363,"user_tz":-420,"elapsed":8,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"1AjM2Gz0qTnE","execution_count":null,"outputs":[]},{"cell_type":"code","source":["! pip install Biopython\n","! sudo apt-get install -y clustalw\n","#! chmod +x clustalw2"],"metadata":{"collapsed":true,"id":"S5kEMtCKzh2A","executionInfo":{"status":"aborted","timestamp":1727709424363,"user_tz":-420,"elapsed":8,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"S5kEMtCKzh2A","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## **Import**"],"metadata":{"id":"YmLdVI1y0aiL"},"id":"YmLdVI1y0aiL"},{"cell_type":"code","source":["from Bio import Entrez, SeqIO\n","from Bio.Blast import NCBIWWW, NCBIXML\n","from Bio.Align.Applications import ClustalwCommandline\n","import os"],"metadata":{"id":"o768uoGM0eb-","executionInfo":{"status":"aborted","timestamp":1727709424363,"user_tz":-420,"elapsed":8,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"o768uoGM0eb-","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["# **0. FETCH FASTA SEQUENCE FROM PUBLIC DATABASE (NCBI, GENBANK, GISAID)**"],"metadata":{"id":"uGvYtlKbe8Bp"},"id":"uGvYtlKbe8Bp"},{"cell_type":"markdown","source":["## **0A. Fetching a sequence (ID) from GenBank/NCBI using biopython**"],"metadata":{"id":"BVjtPR68fRtz"},"id":"BVjtPR68fRtz"},{"cell_type":"code","source":["from Bio import Entrez, SeqIO\n","\n","# input email sebagai identitas\n","Entrez.email = \"your-email@example.com\"\n","\n","# Fetch sekuen dengan menggunakan ACCESSION NUMBER ID GENBANK\n","def fetch_sequence_from_genbank(accession_id):\n","    handle = Entrez.efetch(db=\"nucleotide\", id=accession_id, rettype=\"gb\", retmode=\"text\")\n","    record = SeqIO.read(handle, \"genbank\")\n","    handle.close()\n","\n","    # Print the sequence\n","    print(f\"Description: {record.description}\")\n","    print(f\"Sequence: {record.seq}\")\n","    #print(f\"Sequence: {record.annotations}\")\n","\n","    return record\n","\n","# Example usage\n","genbank_record = fetch_sequence_from_genbank(\"AB425198.1\")  # Example: esophageal cancer protein 1 genome"],"metadata":{"id":"lGeW9ZcvfgsL","executionInfo":{"status":"aborted","timestamp":1727709424363,"user_tz":-420,"elapsed":7,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"lGeW9ZcvfgsL","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### check hieracycal data structure"],"metadata":{"id":"ihl0aA58-IVj"},"id":"ihl0aA58-IVj"},{"cell_type":"code","source":["# annotations field; dic that provides a number of properties for the seq\n","print('Annotations dictionary:\\n')\n","print(genbank_record.annotations)\n","\n","print('\\nKeys:')\n","print(genbank_record.annotations.keys())"],"metadata":{"id":"eZeI80r4XBBj","executionInfo":{"status":"aborted","timestamp":1727709424364,"user_tz":-420,"elapsed":8,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"eZeI80r4XBBj","execution_count":null,"outputs":[]},{"cell_type":"code","source":["# Getting specific key values\n","gb=genbank_record\n","print('\\nGet specific parts of the annotation:\\n')\n","print('Taxonomy:\\n')\n","print(gb.annotations['taxonomy'])\n","\n","print('Source:\\n')\n","print(gb.annotations['source'])\n","\n","print('Date:\\n')\n","print(gb.annotations['date'])\n","\n","#print('gi number:\\n')\n","#print(gb.annotations['gi'])"],"metadata":{"id":"dypdA5-iXx_7","executionInfo":{"status":"aborted","timestamp":1727709424364,"user_tz":-420,"elapsed":8,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"dypdA5-iXx_7","execution_count":null,"outputs":[]},{"cell_type":"code","source":["gb.features[:5] # show only a few features (total=41)"],"metadata":{"id":"euvZIHJucpH2","executionInfo":{"status":"aborted","timestamp":1727709424364,"user_tz":-420,"elapsed":8,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"euvZIHJucpH2","execution_count":null,"outputs":[]},{"cell_type":"code","source":["gene_features = []\n","for i in range(len(gb.features)):\n","    if(gb.features[i].type == 'gene'):\n","        gene_features.append(gb.features[i])\n","\n","print(f'Number of gene features: {len(gene_features)}')\n","gene_features"],"metadata":{"id":"afZ3zgj_ctqN","executionInfo":{"status":"aborted","timestamp":1727709424364,"user_tz":-420,"elapsed":8,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"afZ3zgj_ctqN","execution_count":null,"outputs":[]},{"cell_type":"code","source":["# Gene Qualifiers\n","gene_features[0].qualifiers"],"metadata":{"id":"ScG1_DI9cyeE","executionInfo":{"status":"aborted","timestamp":1727709424364,"user_tz":-420,"elapsed":7,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"ScG1_DI9cyeE","execution_count":null,"outputs":[]},{"cell_type":"code","source":["CDS_features = []\n","for i in range(len(gb.features)):\n","    if(gb.features[i].type == 'CDS'):\n","        CDS_features.append(gb.features[i])\n","\n","print(f\"Number of CDS features: {len(CDS_features)}\")\n","CDS_features"],"metadata":{"id":"kyXjGcJ-c3ZQ","executionInfo":{"status":"aborted","timestamp":1727709424364,"user_tz":-420,"elapsed":6,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"kyXjGcJ-c3ZQ","execution_count":null,"outputs":[]},{"cell_type":"code","source":["print(f'CDS Qualifier Keys: {CDS_features[0].qualifiers.keys()}\\n')\n","\n","print('Showing First CDS Feature')\n","print(CDS_features[0].qualifiers) # ordered dictionary"],"metadata":{"id":"i-cqU8T_d_gL","executionInfo":{"status":"aborted","timestamp":1727709424364,"user_tz":-420,"elapsed":6,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"i-cqU8T_d_gL","execution_count":null,"outputs":[]},{"cell_type":"code","source":["for key, value in CDS_features[0].qualifiers.items():\n","    print(f'{key} : {value}')"],"metadata":{"id":"GgrbGzWSeFBR","executionInfo":{"status":"aborted","timestamp":1727709424364,"user_tz":-420,"elapsed":6,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"GgrbGzWSeFBR","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## **0B. Fetching Multiple sequences by query from NCBI**"],"metadata":{"id":"96aBR6cxf8Ov"},"id":"96aBR6cxf8Ov"},{"cell_type":"code","source":["from Bio import Entrez, SeqIO\n","\n","Entrez.email = \"your-email@example.com\"\n","\n","# Function to fetch multiple sequences from GenBank based on a search query\n","def fetch_multiple_sequences(query, max_records):\n","\n","    # Search for nucleotide sequences matching the query\n","    search_handle = Entrez.esearch(db=\"nucleotide\", term=query, retmax=max_records)\n","    search_results = Entrez.read(search_handle)\n","    search_handle.close()\n","\n","    ids = search_results[\"IdList\"]\n","\n","    # Fetch the sequences in FASTA format\n","    fetch_handle = Entrez.efetch(db=\"nucleotide\", id=ids, rettype=\"fasta\", retmode=\"text\")\n","    sequences = list(SeqIO.parse(fetch_handle, \"fasta\"))\n","    fetch_handle.close()\n","\n","    # Print the total number of sequences fetched\n","    print(f\"Fetched {len(sequences)} sequences\")\n","\n","    # Optionally save to a file\n","    with open(\"sequences.fasta\", \"w\") as output_handle:\n","        SeqIO.write(sequences, output_handle, \"fasta\")\n","\n","    return sequences\n","\n","# Example usage: Search for BRCA1 gene in Homo sapiens\n","sequences = fetch_multiple_sequences(\"BRCA1[Gene] AND Homo sapiens[Organism]\", max_records=50)"],"metadata":{"id":"EHl-ZC5igNNM","executionInfo":{"status":"aborted","timestamp":1727709424364,"user_tz":-420,"elapsed":6,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"EHl-ZC5igNNM","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## **0C. Reading sequences from local FASTA file**"],"metadata":{"id":"fFDkAzO4hHxB"},"id":"fFDkAzO4hHxB"},{"cell_type":"code","source":["from Bio import SeqIO\n","\n","# Function to read sequences from a FASTA file\n","def read_fasta(file_path):\n","    sequences = list(SeqIO.parse(file_path, \"fasta\"))\n","    print(f\"Found {len(sequences)} sequences in {file_path}\")\n","\n","    for seq_record in sequences[:5]:  # Print the first 5 sequences\n","        print(f\"ID: {seq_record.id}\")\n","        print(f\"Sequence: {seq_record.seq[:100]}...\")  # Print the first 100 bases\n","\n","    return sequences\n","\n","# Example usage\n","fasta_sequences = read_fasta(\"sequences.fasta\")"],"metadata":{"id":"C-tdEcD_hUnn","executionInfo":{"status":"aborted","timestamp":1727709424365,"user_tz":-420,"elapsed":7,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"C-tdEcD_hUnn","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## **0D. Akses NCBI Virus Data Hub**"],"metadata":{"id":"GBcavXCeiPvF"},"id":"GBcavXCeiPvF"},{"cell_type":"code","source":["def fetch_viral_sequences(query, max_records):\n","    Entrez.email = \"your-email@example.com\"\n","\n","    # Search for viral sequences\n","    search_handle = Entrez.esearch(db=\"nucleotide\", term=query, retmax=max_records)\n","    search_results = Entrez.read(search_handle)\n","    search_handle.close()\n","\n","    ids = search_results[\"IdList\"]\n","\n","    # Fetch the sequences in FASTA format\n","    fetch_handle = Entrez.efetch(db=\"nucleotide\", id=ids, rettype=\"fasta\", retmode=\"text\")\n","    sequences = list(SeqIO.parse(fetch_handle, \"fasta\"))\n","    fetch_handle.close()\n","\n","    # Save to a file\n","    with open(\"viral_sequences.fasta\", \"w\") as output_handle:\n","        SeqIO.write(sequences, output_handle, \"fasta\")\n","\n","    return sequences\n","\n","# Example usage: Fetching SARS-CoV-2 sequences\n","viral_sequences = fetch_viral_sequences(\"SARS-CoV-2[Organism]\", max_records=100)"],"metadata":{"id":"61GT3284ifo7","executionInfo":{"status":"aborted","timestamp":1727709424365,"user_tz":-420,"elapsed":7,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"61GT3284ifo7","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## **0E. Downloading Data in Bulk with Biophyton**"],"metadata":{"id":"LRhuGEJ6i3GI"},"id":"LRhuGEJ6i3GI"},{"cell_type":"code","source":["# Download a large number of records (e.g., 1000 sequences) in batches of 200\n","def batch_download_sequences(query, batch_size=200, total_records=1000):\n","    Entrez.email = \"your-email@example.com\"\n","\n","    for start in range(0, total_records, batch_size):\n","        handle = Entrez.esearch(db=\"nucleotide\", term=query, retstart=start, retmax=batch_size)\n","        record = Entrez.read(handle)\n","        ids = record[\"IdList\"]\n","\n","        # Fetch sequences\n","        handle = Entrez.efetch(db=\"nucleotide\", id=ids, rettype=\"fasta\", retmode=\"text\")\n","        data = handle.read()\n","        handle.close()\n","\n","        with open(f\"sequences_batch_{start}.fasta\", \"w\") as f:\n","            f.write(data)\n","\n","        print(f\"Downloaded {len(ids)} records (Batch {start // batch_size + 1})\")\n","\n","# Example: Downloading large viral dataset\n","batch_download_sequences(\"SARS-CoV-2[Organism]\", batch_size=200, total_records=1000)"],"metadata":{"id":"6hXIa_zni_Sl","executionInfo":{"status":"aborted","timestamp":1727709424365,"user_tz":-420,"elapsed":7,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"6hXIa_zni_Sl","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## **0F. Downloading Data in Bulk with Biophyton using reference file (.txt)**"],"metadata":{"id":"qxWux_LyyEYY"},"id":"qxWux_LyyEYY"},{"cell_type":"code","source":["def download_cytochrome_b_sequences(species_list):\n","    Entrez.email='researchtirta@gmail.com'\n","\n","    for species in species_list:\n","        search_term = f\"{species} AND mitochondrial cytochrome b gene, complete cds AND 1:2000[SLEN]\"\n","        handle = Entrez.esearch(db='nucleotide', term=search_term, retmax=5)\n","        record = Entrez.read(handle)\n","        handle.close()\n","\n","        if int(record['Count'] == 0):\n","            print(f\"No cytochrome b sequences found for {species}\")\n","        else:\n","            seq_ids = record['IdList']\n","            handle = Entrez.efetch(db='nucleotide', id=seq_ids, rettype='fasta', retmode='text')\n","            sequences = SeqIO.parse(handle,'fasta')\n","            sequence_list = list(sequences)\n","\n","            output_file = f\"{species}_cytb.fasta\"\n","            SeqIO.write(sequence_list, output_file, 'fasta')\n","            handle.close()\n","            num_sequences = len(sequence_list)\n","            print(f\"Downloaded {num_sequences} cytochrome b sequence(s) for {species}\")"],"metadata":{"id":"rbfUUxdRyTrt","executionInfo":{"status":"aborted","timestamp":1727709424365,"user_tz":-420,"elapsed":6,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"rbfUUxdRyTrt","execution_count":null,"outputs":[]},{"cell_type":"code","source":["file_path = 'species_file.txt'\n","\n","with open(file_path, 'r') as file:\n","    species_list = [line.strip() for line in file]\n","\n","download_cytochrome_b_sequences(species_list)"],"metadata":{"id":"kFyi8kO7y0Cv","executionInfo":{"status":"aborted","timestamp":1727709424365,"user_tz":-420,"elapsed":6,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"kFyi8kO7y0Cv","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## **0G. Combine or merge multiple fasta file**"],"metadata":{"id":"D-2bDOhwj4in"},"id":"D-2bDOhwj4in"},{"cell_type":"code","source":["from Bio import SeqIO\n","\n","# List of input FASTA files you want to combine\n","fasta_files = [\"sequences_batch_0.fasta\", \"sequences_batch_200.fasta\",\n","               \"sequences_batch_400.fasta\", \"sequences_batch_600.fasta\",\n","               \"sequences_batch_800.fasta\"]\n","\n","# Name of the output file\n","output_file = \"combined_sequences.fasta\"\n","\n","# Open the output file for writing\n","with open(output_file, \"w\") as outfile:\n","    for fasta_file in fasta_files:\n","        # Parse each input FASTA file\n","        with open(fasta_file, \"r\") as infile:\n","            # Use SeqIO.parse to read sequences and write them to the output file\n","            for record in SeqIO.parse(infile, \"fasta\"):\n","                SeqIO.write(record, outfile, \"fasta\")\n","\n","print(f\"Sequences from {len(fasta_files)} files have been combined into {output_file}\")"],"metadata":{"id":"8VcAsFJ2kEBx","executionInfo":{"status":"aborted","timestamp":1727709424365,"user_tz":-420,"elapsed":6,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"8VcAsFJ2kEBx","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## **LATIHAN**"],"metadata":{"id":"alvKQJzcOW8B"},"id":"alvKQJzcOW8B"},{"cell_type":"markdown","source":["### Tugas"],"metadata":{"id":"sOCyBJg4aaZ8"},"id":"sOCyBJg4aaZ8"},{"cell_type":"markdown","source":["1. Silahkan buat analisis sekuen dengan ID: NC_045512:\n","\n","    a. Temukan Key dan Value dari ID tersebut\n","\n","    b. Jelaskan informasi terkait sekuen dengan ID tersebut\n","\n","2. Downloadlah dataset variant dari covid-19 informasi ada disini https://en.wikipedia.org/wiki/Variants_of_SARS-CoV-2:\n","\n","    a. Downloadlah menggunakan file referensi dengan format .txt\n","\n","    b. Gunakan fungsi untuk melakukan merger file dengan format **(fasta dan ganbank)**\n","\n","\n","3. Tugas dikumpulkan dalam format nim_nama.ipynb (untuk kodingan) dan nim_nama.fasta & nim_nama.gb pada GCR sebelum 1 Oktober 2024, pukul 14.30\n"],"metadata":{"id":"C4X8hsJiagAS"},"id":"C4X8hsJiagAS"},{"cell_type":"code","source":[],"metadata":{"id":"eTN_v-xXSz4u","executionInfo":{"status":"aborted","timestamp":1727709424365,"user_tz":-420,"elapsed":6,"user":{"displayName":"065_MELIZA WULANDARI","userId":"15457601190170839248"}}},"id":"eTN_v-xXSz4u","execution_count":null,"outputs":[]}],"metadata":{"kernelspec":{"display_name":"Python 3 (ipykernel)","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.10.5"},"colab":{"provenance":[],"collapsed_sections":["uGvYtlKbe8Bp","BVjtPR68fRtz","96aBR6cxf8Ov","fFDkAzO4hHxB","GBcavXCeiPvF","LRhuGEJ6i3GI","qxWux_LyyEYY","D-2bDOhwj4in","alvKQJzcOW8B","188b8cbb-d41b-4ee0-bde7-9cc2fb85e49f","6b29a436-a75b-43bc-b6d2-7cc35788bf54","a3eb20ae-4824-495f-acf8-227a500134a0","d5d25134-d48d-44c2-8f24-356600400d9c","edb45e35-b3bb-4f9a-a7d8-71bb5504a07b","1e09f6e1-1f16-4ecd-928e-a61a0d6226e0","c7277b60-c1b7-4e7a-9584-1000e411d2f4","2be4e53a-decb-4cc4-930d-e2627d5f89e1","14458c34-d00d-462c-9657-f4404711db92","f52d9911-64dc-4f2c-addc-dc4a08d0c623","f40f8bcf-43ad-47c2-9316-683312260a72","4999391e-2455-44c9-b84b-5398db853fd6","5xWQbMK6lMIR","nyfDTKO70NYN","2KewAAfR1AuX","1BOjr7_cozWt","IX_QuS6XsxgV","7bzIrD0VvcZ5","Ii8MeieJ_aMw","VMgKDdvGpHXQ","Q1pBNPzjUtii","fsqo5e9Lv4sM"]}},"nbformat":4,"nbformat_minor":5}